django/thunderstore/repository/tasks/markdown.py

```diff
@@ -0,0 +1,35 @@
+from celery import shared_task
+
+from thunderstore.cache.utils import get_cache
+from thunderstore.core.settings import CeleryQueues
+from thunderstore.markdown.templatetags.markdownify import render_markdown
+
+DEFAULT_CACHE_EXPIRY = 60 * 5
+CACHE_LOCK_TIMEOUT = 30
+cache = get_cache("markdown_render")
+
+
+@shared_task(
+    queue=CeleryQueues.BackgroundMarkdownRender,
+    name="thunderstore.repository.tasks.render_markdown",
+)
+def render_markdown_to_html(markdown: str, type: str, package_version_id: int) -> str:
+    cache_key = f"rendered_html:{type}:{package_version_id}"
+    lock_key = f"lock.rendered_html:{type}:{package_version_id}"
+    status_key = f"rendering_status:{type}:{package_version_id}"
+
+    cached_html = cache.get(cache_key)
+    if cached_html is not None:
+        return cached_html
+
+    with cache.lock(lock_key, timeout=CACHE_LOCK_TIMEOUT, blocking_timeout=None):
+        try:
+            html = render_markdown(markdown)
+        except Exception as error:
+            cache.delete(status_key)
+            raise error
+
+        cache.set(cache_key, html, timeout=DEFAULT_CACHE_EXPIRY)
```

Comment:
This cache can be permanent, no need to expire immutable data.

That said, if the markdown renderer code changes we _could_ bust the cache if we have a reasonable mechanism for detecting that somehow. I'd not consider it a huge issue even if we don't bust the cache in that case though, as long as we document it somewhere.